backend/src/openarchiefbeheer/destruction/models.py

```diff
@@ -235,33 +235,38 @@ def generate_destruction_report(self) -> None:
             "zaaktype omschrijving",
             "selectielijst procestype nummer",
         ]
-        with NamedTemporaryFile(mode="w", newline="", delete_on_close=False) as f_tmp:
-            writer = csv.DictWriter(f_tmp, fieldnames=fieldnames)
-            writer.writeheader()
-            for item in self.items.filter(
-                processing_status=InternalStatus.succeeded
-            ).iterator(chunk_size=1000):
-                data = {
-                    **item.extra_zaak_data,
-                    **{
-                        "zaaktype url": item.extra_zaak_data["zaaktype"]["url"],
-                        "zaaktype omschrijving": item.extra_zaak_data["zaaktype"][
-                            "omschrijving"
-                        ],
-                        "selectielijst procestype nummer": item.extra_zaak_data[
-                            "zaaktype"
-                        ]["selectielijst_procestype"]["nummer"],
-                    },
-                }
-                del data["zaaktype"]
-
-                writer.writerow(data)
+        with NamedTemporaryFile(mode="wb", delete_on_close=False) as f_tmp:
+            workbook = xlsxwriter.Workbook(f_tmp.name, options={"in_memory": False})
+            worksheet = workbook.add_worksheet(name=gettext("Deleted zaken"))
+            worksheet.write_row(0, 0, fieldnames)
+
+            for row_count, item in enumerate(
+                self.items.filter(processing_status=InternalStatus.succeeded).iterator(
+                    chunk_size=1000
+                )
+            ):
+                data = [
+                    item.extra_zaak_data["url"],
```

Comment:
This looks a lot like the fieldnames variable, can we somehow map them so we can maintain the fields in one place?